feat: update the RBAC documentation to match the new implementation (#1728)
mendersoftware/mender-docs: 02.Overview/12.Role.Based.Access.Control/docs.md

CI/CD systems), this Role can only manage Artifacts, including uploading new Artifacts
and deleting existing Artifacts. It does not have access to Devices or Deployments.

* **Deployments Manager:** Intended for Users responsible for managing deployments,
this Role can create and abort deployments for all the devices.

* **Troubleshooting:** Access to the troubleshooting features for all the devices:
Remote Terminal, File Transfer, Port Forwarding.

## Custom Roles and Permissions

You can create new Roles based on your specific needs and workflows. When creating or
editing custom Roles, you can specify the Role's name, description and a set of
permissions to grant to the Users bound to the Role.

Some of the permissions apply at the tenant level:

* **User Management**: these permissions should be granted carefully, as they
allow privilege increases for any Users managed by a User with User management
permissions.
  * **Read**: read access to the User management sections.
  * **Manage**: write access to the User management sections, requires **Read**.

* **System audit log**: these permissions allow tracing changes to devices,

> releases and User accounts, as well provide information about deployments.

Reviewer: ```suggestion
releases and User accounts, as well as providing information about deployments.
```

>   * **Read**: read access to the audit logs sections.

Reviewer: ```suggestion
  * **Read**: read access to the audit log section.
```
This is probably rather subjective, as we have only one auditlog so far, but the details etc. should allow talking about _sections_ - so 🤷‍♂️ whatever you prefer.

* **Releases**: these permissions can be granted to allow artifact and release
modifications, as well as the creation of new releases.
  * **Read**: read access to the releases and artifacts sections.

>   * **Upload**: upload-only access to the releases and artifacts logs sections.
>   * **Manage**: full write access to the releases and artifacts logs sections, requires **Read** and **Upload**.

Reviewer: ```suggestion
  * **Upload**: upload-only access to the releases and artifacts sections.
  * **Manage**: full write access to the releases and artifacts sections, requires **Read** and **Upload**.
```

The device-specific permissions can apply to either all the devices or selected
static groups of devices:

* **Read**: read access to the devices.
* **Manage**: write access to the devices, requires **Read**.
* **Deploy**: permission to deploy software to the devices, requires **Read**.

> * **Connect**: troubleshooting access to the devices, requires **Read**.

Reviewer: Here I'm not sure if the implications are clear enough... In the GUI I didn't want to mark the **Manage** right as required, since this is still a bit broader, but connecting to the device does also give write access, although "only" limited to what you can do on the live device, so likely not a full system update in the way **Manage** allows (unless the configured user is able to download & stand-alone install an update)...
Maybe:
_troubleshooting access to the device, also allows write access to the devices, requires **Read**_
but I'm not sure how detailed we should be here...

Author: I added this sentence:

```
!!! Please note that the **Connect** permission gives users access to the Remote Terminal,
!!! File Transfer, and Port-forwarding features. Through these, it is possible to alter
!!! the devices locally, for example, changing the Mender Client configuration.
!!! Therefore, you can consider it a form of write access to the devices.
```
